schematic/utils/df_utils.py

```diff
@@ -77,6 +77,10 @@ def update_df(
     input_df_idx.reset_index(inplace=True)
     input_df_idx = input_df_idx[input_df.columns]
 
+     # Sometimes pandas update can change the column datatype, recast
+    for col in input_df_idx.columns:
+        input_df_idx[col] = input_df_idx[col].astype(input_df.dtypes[col])
```

Comment:
@mialy-defelice just confirming I understand correctly, the dtypes gets the numpy type of data in the given column and astype ensures pandas sets up the type accordingly in the dataframe?

Reply:
@milen-sage It was converting int to float. So yes at the end of the transformations, I just use as type to reset the type to the column types in the input table.

Reply:
I'll go through later and see why all those tests are failing.

Reply:
got it